awx/main/tests/functional/test_named_url.py

```diff
@@ -0,0 +1,244 @@
+import pytest
+
+from awx.api.versioning import reverse
+from awx.main.middleware import URLModificationMiddleware
+from awx.main.models import * # noqa
+
+
+@pytest.mark.django_db
+def test_user(get, admin_user):
+    # In real-world scenario, named url graph structure is populated by __init__
+    # of URLModificationMiddleware. The way Django bootstraps ensures the initialization
+    # will happen *once and only once*, while the number of initialization is uncontrollable
+    # in unit test environment. So it is wrapped by try-except block to mute any
+    # unwanted exceptions.
+    try:
+        URLModificationMiddleware()
+    except Exception:
+        pass
+    test_user = User.objects.create(username='test_user', password='test_user', is_superuser=False)
+    url = reverse('api:user_detail', kwargs={'pk': test_user.pk})
+    response = get(url, user=admin_user, expect=200)
+    assert response.data['related']['named_url'].split('/')[-2] == 'test_user'
```

Comment:
It looks like you're using this `.split('/')[-2]` to get the end of the string.  Would `endswith()` be simpler for most of these cases?

```python
assert "/api/v2/users/ryan/".endswith("/ryan/")
```